src/pages/studyView/addChartButton/genericAssaySelection/GenericAssaySelection.tsx

```diff
@@ -0,0 +1,290 @@
+import * as React from 'react';
+import _ from 'lodash';
+import { GenericAssayChart } from 'pages/studyView/StudyViewPageStore';
+import { observer } from 'mobx-react';
+import autobind from 'autobind-decorator';
+import { action, computed, observable, toJS } from 'mobx';
+import ReactSelect from 'react-select';
+import { deriveDisplayTextFromGenericAssayType } from 'pages/resultsView/plots/PlotsTabUtils';
+import { MolecularProfileOption } from 'pages/studyView/StudyViewUtils';
+
+export interface IGenericAssaySelectionProps {
+    molecularProfileOptions: MolecularProfileOption[];
+    genericAssayEntityOptions: ISelectOption[];
+    genericAssayType: string;
+    submitButtonText: string;
+    onSubmit: (charts: GenericAssayChart[]) => void;
+}
+
+interface ISelectOption {
+    value: string;
+    label: string;
+}
+
+@observer
+export default class GenericAssaySelection extends React.Component<
+    IGenericAssaySelectionProps,
+    {}
+> {
+    @observable private _selectedProfileOption?: MolecularProfileOption & {
+        profileName: string;
+    };
+
+    @observable.ref private _selectedGenericAssayEntityIds: string[] = [];
+    @observable private _genericAssaySearchText: string = '';
+
+    @autobind
+    @action
+    private onAddChart() {
+        if (this.selectedProfileOption !== undefined) {
+            const charts = this._selectedGenericAssayEntityIds.map(entityId => {
+                return {
+                    name:
+                        entityId +
+                        ': ' +
+                        this.selectedProfileOption!.profileName,
+                    description: this.selectedProfileOption!.description,
+                    profileType: this.selectedProfileOption!.value,
+                    genericAssayType: this.props.genericAssayType,
+                    dataType: this.selectedProfileOption!.dataType!,
+                    genericAssayEntityId: entityId,
+                };
+            });
+            this.props.onSubmit(charts);
+        }
+    }
+
+    @autobind
+    @action
+    private handleProfileSelect(option: any) {
+        if (option && option.value) {
+            this._selectedProfileOption = option;
+        }
+    }
+
+    @computed
+    private get selectedProfileOption() {
+        if (this._selectedProfileOption !== undefined) {
```

Comment:
should really be if else else.   i think it's better to get in habit of writing out conditionals.